### PR TITLE
Move from ant to maven

### PR DESCRIPTION
**Pull request by rvennam**

### CODE REVIEW

File: src/main/webapp/util.js [hunk 1]
```diff
@@ -103,15 +103,15 @@ function xhrDelete(url, callback, errback){
...
+	return window.JSON ? JSON.parse(str) : eval('(' + str + ')');
 }
 
 function objectToQuery(map){
 	var enc = encodeURIComponent, pairs = [];
 	for(var name in map){
 		var value = map[name];
 		var assign = enc(name) + "=";
-		if(value && value instanceof Array){
+		if(value && (value instanceof Array || typeof value == 'array')){
```
Comment (degrim) [inline src/main/webapp/util.js]:
```
(function() {
var arr = [];
alert(typeof arr == "array");
})();
```

Output of this function is false, therefore the typeof check here will always be false.

File: src/main/webapp/util.js [hunk 2]
```diff
@@ -103,15 +103,15 @@ function xhrDelete(url, callback, errback){
 }
 
 function parseJson(str){
-	return JSON.parse(str);
+	return window.JSON ? JSON.parse(str) : eval('(' + str + ')');
```
Comment (degrim) [inline src/main/webapp/util.js]: Evaluating untrusted strings as JavaScript code just seems like a bad idea, any reason to change this back?

File: src/main/webapp/util.js [hunk 3]
```diff
@@ -17,8 +17,8 @@ function xhrGet(url, callback, errback){
 	var xhr = new createXHR();
 	xhr.open("GET", url, true);
 	xhr.onreadystatechange = function(){
-		if(xhr.readyState === 4){
-			if(xhr.status === 200){
+		if(xhr.readyState == 4){
```
Comment (degrim) [inline src/main/webapp/util.js]: === is more specific in that it checks the type of the operand as well as the value.  See [this SO post](http://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons).  Was this causing some issue?

File: src/main/java/example/nosql/CloudantClientMgr.java [hunk 4]
```diff
@@ -19,8 +19,8 @@
 
 	private static String databaseName = "sample_nosql_db";
 
-	private static String user = null;
-	private static String password = null;
+	private static String user = "REPLACE_WITH_CLOUDANT_USERNAME";
```
Comment (degrim) [inline src/main/java/example/nosql/CloudantClientMgr.java]: I know this isn't new, but would be nice if we could read the credentials from a file when running locally like the other cloudant boilerplates do so we're not asking users to put their credentials in source code.

**Comment by opiethehokie** [comment]
Thanks for doing this. A few questions/comments ...

* What are those dependencies? I can make it build with these which seems simpler:

```
	<dependencies>
		<dependency>
			<groupId>javax.servlet</groupId>
			<artifactId>javax.servlet-api</artifactId>
			<version>3.0.1</version>
			<scope>provided</scope>
		</dependency>
		<dependency>
			<groupId>javax.ws.rs</groupId>
			<artifactId>javax.ws.rs-api</artifactId>
			<version>2.0.1</version>
			<scope>provided</scope>
		</dependency>
		<dependency>
			<groupId>com.cloudant</groupId>
			<artifactId>cloudant-client</artifactId>
			<version>2.7.0</version>
		</dependency>
		<dependency>
			<groupId>com.google.code.gson</groupId>
			<artifactId>gson</artifactId>
			<version>2.8.0</version>
		</dependency>
	</dependencies>
```

* There's still jars in WEB-INF/lib when we can get them from maven
* There are 2 WEB-INF directories
* Why add SimpleServlet.java here?
* Is src/main/wlp/server.xml needed?

* I can build the war fine in eclipse, but otherwise would agree with @degrim's other comments

File: .gitignore [hunk 5]
```diff
@@ -1 +1,4 @@
-bin
+target/*
+|target/*.war
```
Comment (opiethehokie) [inline .gitignore]: I'd like to keep the WAR in git for now.

File: src/main/webapp/util.js (hunk 3 above)
Comment (rvennam) [inline src/main/webapp/util.js]: I accidently regressed this. fixed

File: src/main/webapp/util.js (hunk 2 above)
Comment (rvennam) [inline src/main/webapp/util.js]: I accidently regressed this. fixed

File: src/main/webapp/util.js (hunk 1 above)
Comment (rvennam) [inline src/main/webapp/util.js]: I accidently regressed this. fixed

File: src/main/java/example/nosql/CloudantClientMgr.java (hunk 4 above)
Comment (rvennam) [inline src/main/java/example/nosql/CloudantClientMgr.java]: Completely agree - I'm going to make a new issue for this.

**Comment by rvennam** [comment]
@degrim From Eclipse, you can right click on the pom.xml and select Run As -> Maven install. You can also run `mvn clean install` from the command line - I've added this to the README.

@opiethehokie 
1. The dependencies in the pom.xml are recommended dependencies from WAS app accelerator. I would like to keep it consistent. You need them and the server.xml to run using `mvn liberty:run-server`. I've added this instruction to the README.
2. I removed the libs
3. I removed one of the WEB-INF dir

**Comment by rvennam** [comment]
@opiethehokie @degrim Thanks for the review - See updates/comments.

**Comment by rvennam** [comment]
@opiethehokie fixed the attachment lookup

**Pull request merged**